libs/authtoken/authtoken.go

```diff
@@ -3,21 +3,21 @@ package authtoken
 import (
 	"encoding/json"
 
-	"github.com/cristalhq/jwt"
+	"github.com/cristalhq/jwt/v5"
 	"github.com/filecoin-project/go-jsonrpc/auth"
 
 	"github.com/celestiaorg/celestia-node/api/rpc/perms"
 )
 
 // ExtractSignedPermissions returns the permissions granted to the token by the passed signer.
 // If the token isn't signed by the signer, it will not pass verification.
-func ExtractSignedPermissions(signer jwt.Signer, token string) ([]auth.Permission, error) {
-	tk, err := jwt.ParseAndVerifyString(token, signer)
+func ExtractSignedPermissions(verifier jwt.Verifier, token string) ([]auth.Permission, error) {
+	tk, err := jwt.Parse([]byte(token), verifier)
 	if err != nil {
 		return nil, err
 	}
 	p := new(perms.JWTPayload)
-	err = json.Unmarshal(tk.RawClaims(), p)
+	err = json.Unmarshal(tk.Claims(), p)
```

Comment:
`Claims` returns `json.RawMessage` which is exactly what was returned by `RawClaims`.